unlock-app/src/components/interface/user-account/PaymentDetails.tsx

```diff
@@ -31,19 +35,54 @@ export const PaymentDetails = React.memo(
   }
 )
 
-interface PaymentFormProps {
-  signPaymentData: (stripeTokenId: string) => any
-}
-
 export class PaymentForm extends React.Component<
   PaymentFormProps & ReactStripeElements.InjectedStripeProps
 > {
+  constructor(props: any) {
+    super(props)
+    this.state = {
+      cardHolderName: '',
+      address1: '',
+      address2: '',
+      addressCity: '',
+      addressState: '',
+      addressCountry: '',
+    }
+  }
+
+  handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value, name } = e.target
+    this.setState(prevState => {
+      const newState = {
```

Comment:
Let's use the non-callback form of setState since it does not depend on the old value:
```js
this.setState({ [name]: value })
```

This will merge with the existing state